index.html

```diff
@@ -272,6 +272,28 @@ <h3>
       The <dfn data-dfn-for="DigitalCredential">data</dfn> member is the
       credential's response data.
     </p>
+    <h3>
+      [[\DiscoverFromExternalSource]](origin, options, sameOriginWithAncestors)
+      internal method
+    </h3>
+    <p>
+      When invoked, the <dfn class="export" data-dfn-for=
+      "DigitalCredential">[[\DiscoverFromExternalSource]](origin, options,
+      sameOriginWithAncestors)</dfn> internal method MUST:
+    </p>
+    <li>Let |global| be [=this=]'s [=relevant global object=].
+    </li>
+    <li>Let |document| be |global|'s [=associated `Document`=].
+    </li>
+    <li>If |document| is not a [=Document/fully active descendant of a
+    top-level traversable with user attention=], [=exception/throw=]
+    {{"NotAllowedError"}} {{DOMException}}.
+    </li>
+    <li>If |window| does not have [transient activation], [=exception/throw=]
+    {{"NotAllowedError"}} {{DOMException}}.
+    </li>
+    <li>[=Consume user activation=] of |window|.
+    </li>
```

Comment:
This algorithm makes general sense up until this point, but for it to be type-correct, it has to return a Credential, right, since [DiscoverFromExternalSource](https://www.w3.org/TR/credential-management-1/#algorithm-discover-creds) returns a `DigitalCredential`?

While we are still figuring out the details of the algorithm from this point, but we could leave for now a TODO here to complete this algorithm or something general like "return a DigitalCredential from a wallet"?

Reply:
Yes... we are not quite there yet lol. I'll add a note because it's confusing right now.

Reply:
Also, will switch up the terminology to talk about a "wallet".